content/blog/clickhouse-benchmarking.md

```diff
@@ -0,0 +1,188 @@
+- At **66 million logs**, selected rows spiked to **365,589 rows/sec**, highlighting the system’s ability to maintain high read throughput even as write loads scale.
+
+#### 9. **Inserted Rows per Second (avg_inserted_rows_per_second)**
+
+- The **inserted rows per second** remained consistent at **238,666 rows/sec** for both **1 million and 5 million logs**.
+- During the ingestion of **10 million logs**, the rate climbed to **140,193 rows/sec**, indicating that ClickHouse scaled write operations efficiently.
+- At **66 million logs**, the system handled an impressive **365,589 rows/sec**, reflecting excellent scaling of insertion throughput.
+
+### Incremental Comparison of Key Metrics
+
+| **Logs Ingested** | **Increase in CPU Usage** | **Increase in Selected Bytes/sec** | **Increase in IO Wait (s)** | **Increase in CPU Wait (s)** | **Increase in Disk Reads (B)** | **Increase in Filesystem Reads (B)** | **Increase in Memory Tracked (B)** | **Increase in Inserted Rows/sec** |
+|-------------------|---------------------------|------------------------------------|-----------------------------|------------------------------|---------------------------------|-------------------------------------|------------------------------------|------------------------------------|
+| **1M to 5M**      | 0.103 → 0.111             | 27 KB/s → 37 KB/s                  | 0 → 0                       | 0.0217 → 0.0058              | 0 → 0                           | 439 KB → 533 KB                    | 714 MB → 738 MB                    | 238,666 → 238,666                  |
+| **5M to 10M**     | 0.111 → 0.433             | 37 KB/s → 36 MB/s                  | 0 → 0                       | 0.0058 → 0.3297              | 0 → 1,843 MB                    | 533 KB → 3.32 MB                   | 738 MB → 1,030 MB                  | 238,666 → 140,193                  |
+| **10M to 66M**    | 0.433 → 1.179             | 36 MB/s → 122 MB/s                 | 0 → 0.75                    | 0.3297 → 4.4276              | 1,843 MB → 17.33 MB             | 3.32 MB → 14.30 MB                 | 1,030 MB → 2,065 MB                | 140,193 → 365,589                  |
+
+This incremental comparison clearly shows how the system scales with larger data ingestion volumes. The most notable jumps occur when increasing from **5 million to 10 million logs**, where CPU usage, memory, and disk I/O demands increase sharply. Moving to **66 million logs**, there’s another dramatic rise, particularly in **I/O wait**, **CPU wait**, and **inserted rows per second**.
+
+## Performance in Read-Heavy Operations
+
+ClickHouse’s performance during read-heavy operations, including `SELECT`, aggregate, and `JOIN` queries, is critical for applications relying on fast data retrieval. Here, we analyze key system metrics across different configurations: two-node replicas under load balancing and a single-node configuration due to failover.
```

Comment:
Can we add a diagram to show that we're reading/querying the Clickhouse cluster? 

We can do the same with ingestion. A very simple diagram with nodes and LB in front and the client(the person querying, source in case of ingestion) depicted should suffice.